ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Cleaner.java

```diff
@@ -279,10 +292,14 @@ private void clean(CompactionInfo ci, long minOpenTxnGLB, boolean metricsEnabled
       }
     }
   }
+  
+  private interface CleanerTask<T> {
+    T execute() throws Exception;
+  }
 
-  private ValidReaderWriteIdList getValidCleanerWriteIdList(CompactionInfo ci, Table t, ValidTxnList validTxnList)
+  private ValidReaderWriteIdList getValidCleanerWriteIdList(CompactionInfo ci, ValidTxnList validTxnList)
```

Comment:
Seems like an independent change from the patch - it might be better to do these are in a different PR, so here we can focus on the main stuff

Reply:
it's needed for `safe_delete` use case as we won't be able to resolve table object - metadata would be already removed, we should get full tableName from the ci object.